core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/ScriptController.java

```diff
@@ -9,51 +9,77 @@
         this.scriptPostDtoService = scriptPostDtoService;
-        this.scriptDtoService = scriptDtoService;
+        this.scriptDtoPagedResourcesAssembler = scriptDtoPagedResourcesAssembler;
     }
 
     @GetMapping("")
-    public HalMultipleEmbeddedResource<ScriptDto> getAll(@RequestParam(required = false, name = "expand", defaultValue = "") List<String> expansions,
-                                                         @RequestParam(required = false, name = "version") String version) {
-        List<ScriptDto> scripts = scriptDtoService.getAll(expansions, version != null && version.toLowerCase().equals("latest"));
-        return new HalMultipleEmbeddedResource<>(scripts);
+    public PagedModel<ScriptDto> getAll(Pageable pageable,
+                                        @RequestParam(required = false, name = "expand") List<String> expansions,
+                                        @RequestParam(required = false, name = "version") String version) {
+        Page<ScriptDto> scriptDtoPage = scriptDtoService.getAll(pageable, expansions, version != null && version.toLowerCase().equals("latest"));
+        if (scriptDtoPage.hasContent())
+            return scriptDtoPagedResourcesAssembler.toModel(scriptDtoPage, scriptDtoModelAssembler::toModel);
+        //noinspection unchecked
+        return (PagedModel<ScriptDto>) scriptDtoPagedResourcesAssembler.toEmptyModel(scriptDtoPage, ScriptDto.class);
     }
 
+
     @GetMapping("/{name}")
-    public HalMultipleEmbeddedResource<ScriptDto> getByName(@PathVariable String name, @RequestParam(required = false, name = "expand", defaultValue = "") List<String> expansions) {
-        List<ScriptDto> scripts = scriptDtoService.getByName(name, expansions);
-        return new HalMultipleEmbeddedResource<>(scripts);
+    public PagedModel<ScriptDto> getByName(Pageable pageable,
+                                           @PathVariable String name,
+                                           @RequestParam(required = false, name = "expand") List<String> expansions,
+                                           @RequestParam(required = false, name = "version") String version) {
+        Page<ScriptDto> scriptDtoPage = scriptDtoService.getByName(pageable, name, expansions, version != null && version.toLowerCase().equals("latest"));
+        if (scriptDtoPage.hasContent())
+            return scriptDtoPagedResourcesAssembler.toModel(scriptDtoPage, scriptDtoModelAssembler::toModel);
+        //noinspection unchecked - disable warning on casting
+        return (PagedModel<ScriptDto>) scriptDtoPagedResourcesAssembler.toEmptyModel(scriptDtoPage, ScriptDto.class);
     }
 
     @GetMapping("/{name}/{version}")
-    public ScriptDto get(@PathVariable String name, @PathVariable Long version, @RequestParam(required = false, name = "expand", defaultValue = "") List<String> expansions) throws MetadataDoesNotExistException {
-        return scriptDtoService.getByNameAndVersion(name, version, expansions)
```

Comment:
a default value of "" for a List<> parameters ensures that an empty list is given when no query parameter 'expansions' is provided in the url. Does removing this default value not break any logic in the service or repository?

Reply:
If a null value is passed to the service, it creates an empty value.
Service was already acting that way except that it was doing it through a telescoped constructor.

I'm not sure why I deleted the default value, I think it is because it was giving strange links.
I will check!

Reply:
I put back default value.
I don't remember why I took it off.